pytensor/tensor/rewriting/subtensor_lift.py

```diff
@@ -185,6 +186,63 @@ def local_subtensor_of_elemwise(fgraph, node):
     return [new_out]
 
 
+@register_canonicalize
+@register_specialize
+@node_rewriter([Subtensor])
+def local_subtensor_of_reduce(fgraph, node):
+    """Lift a Subtensor through a CAReduce Op.
+
+    For now rewrite is restricted to single axis of reduction, for simplicity.
+
+    sum(x, axis=1)[0] -> sum(x[0], axis=0)
```

Comment:
Does this rewrite handle the case where `keepdims=True`? Or does that kwarg produce a different `Op` than `CAReduce`?

Reply:
keepdims is not part of CARedue, it's added as a separate expand_dims after the reduction.